sktime/utils/_testing/scenarios.py

```diff
@@ -262,11 +262,8 @@ def _check_dict_of_dict(obj, name="obj"):
     ------
     TypeError if obj is not dict of dict, with str keys
     """
-    msg = f"{obj} must be a dict of dict, with str keys"
-    if not isinstance(obj, dict):
-        raise TypeError(msg)
-    if not np.all([isinstance(x, dict) for x in obj.values()]):
-        raise TypeError(msg)
-    if not np.all([isinstance(x, str) for x in obj.keys()]):
-        raise TypeError(msg)
+    if not (isinstance(obj, dict) and
+            np.all([isinstance(x, dict) for x in obj.values()]) and
+            np.all([isinstance(x, str) for x in obj.keys()])):
```

Comment:
Questions (mainly to @fkiraly I guess):

1. Do we need `np.all` here, is `all` not enough?
2. `all` can do a short circuit directly if I'm not wrong, but you need to skip the internal list. Can we change that?

Reply:
Not sure - I think `all` is enough.

I also think, there's no harm in removing the check entirely, if it is so consequential!
This is so far developer-facing (not even extender-facing) that it doesn't matter if we get rid of it entirely!

Reply:
Are you asking between `np.all` and `all`? My guess is `all` as elements are basic python objects and not `numpy` objects.

Ref. https://stackoverflow.com/a/43382396/11117265

(Probably not much difference though, my comment was mainly from "numpy-not-required-here" perspective.)

Reply:
ok, let's remove numpy then